core/crypto/src/signature.rs

```diff
@@ -255,14 +255,18 @@ impl Hash for PublicKey {
 }
 
 impl Display for PublicKey {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
-        write!(f, "{}", String::from(self))
+    fn fmt(&self, fmt: &mut Formatter) -> std::fmt::Result {
+        let (key_type, key_data) = match self {
+            PublicKey::ED25519(public_key) => (KeyType::ED25519, &public_key.0[..]),
+            PublicKey::SECP256K1(public_key) => (KeyType::SECP256K1, &public_key.0[..]),
+        };
+        fmt.write_fmt(format_args!("{}:{}", key_type, bs58::encode(key_data).into_string()))
```

Comment:
```suggestion
        write!(fmt, "{}:{}", key_type, bs58::encode(key_data).into_string())
```

or am I missing something and there' some beenfit to `.write_fmt` form?

Reply:
Is `into_string` required on `bs58`? Seems like the thing should `impl Dispaly` perhaps, but I wouldn't be surprised if it doesnt'